src/integration/java/io/pinecone/helpers/IndexManager.java

```diff
@@ -41,7 +45,7 @@ private static String findIndexWithDimensionAndPodType(List<String> indexList, i
         while (i < indexList.size()) {
             IndexMeta indexMeta = isIndexReady(indexList.get(i), controlPlaneClient);
             if (indexMeta.getDatabase().getDimension() == dimension
-                    && indexMeta.getDatabase().getPodType().equals("p1.x1")) {
+                    && (indexMeta.getDatabase().getPodType().equals("p1.x1") || config.getEnvironment().equals("gcp-starter"))) {
```

Comment:
Added gcp-starter condition so no new indexes are created if one exists in gcp-starter env. for integration tests.